File: 0000-0999/0065.ValidNumber.py
class Solution(object):
    def isNumber(self, s):
        """
        :type s: str
        :rtype: bool
        """
        s = s.strip()
        if not s:
            return False
        sign = digit = point = e = False
        for ch in s:
            if ch >= '0' and ch <= '9':
                sign = digit = True
            elif ch == '.' and not point:
                sign = point = True
            elif ch in '+-' and not sign and not digit:
                sign = True
            elif ch in 'eE' and not e and digit:
                e = point = True
                sign = digit = False
            else:
                return False
        if digit:
            return True
        else:
            return False
        

File: 0000-0999/0065.ValidNumber.go
func isNumber(s string) bool {
    s = strings.TrimSpace(s)
    num := false
    point := false
    e := false
    for i, ch := range s {
        if ch >= '0' && ch <= '9' {
            num = true
        } else if ch == '.' {
            if point || e {
                return false
            }
            point = true
        } else if ch == 'e' {
            if e || !num {
                return false
            }
            e = true
            num = false
        } else if ch == '+' || ch == '-' {
            if i != 0 && s[i-1] != 'e' {
                return false
            }
        } else {
            return false
        }
    }
    return num
}